prediction/run.py

```diff
@@ -186,4 +208,52 @@ def _balanced_split(self, how_many: int):
         self.y_train = reduce([splitter[i]["y_train"] for i in range(6)])
         self.X_test  = np.array(reduce([splitter[i]["X_test"] for i in range(6)]))
         self.y_test  = reduce([splitter[i]["y_test"] for i in range(6)])
-        
\ No newline at end of file
+        
+    
+def cluster_visualize(data :dict, labels :list):
+    """Visualization of evaluation methods among different number of clusters.
+
+    Args:
+        data (dict): Coniatins result of evaluation scores.
+        labels (list(str)): Ways of representing text.
+    """
+    labels = ["Token", "tfidf", "token freq"]
+    data = {"2cluster" : {"silhouette" : [12, 13, 14], "homogeneity": [12,13,14]},
+    "6cluster" : {"silhouette" : [15, 16, 17], "homogeneity": [15,16,17]},
+    }
+
+    x_pos =  np.arange(len(labels)) 
+    width = 0.35  # the width of the bars
+
+    for key in data["2cluster"].keys():
+        fig, ax = plt.subplots()
+        rects1 = ax.bar(x_pos - width/2, data["2cluster"][key], width, label='2-clusters')
+        rects2 = ax.bar(x_pos + width/2, data["6cluster"][key], width, label='6-clusters')
+
+        # Add some text for labels, title and custom x-axis tick labels, etc.
+        ax.set_ylabel('Scores')
+        ax.set_ylim(top=20)
+        ax.set_title(f'{key} scores by ways of representing text and number of clusters')
+        ax.set_xticks(x_pos)
+        ax.set_xticklabels(labels)
+        ax.legend()
+
+        ax.bar_label(rects1, padding=3)
+        ax.bar_label(rects2, padding=3)
+
+        fig.tight_layout()
+
+    plt.show()
+
+def classify_visualize(accs : list, labels : list):
```

Comment:
And here

Reply:
If it's a visualization per group, it should be reflected in the method's name. Unless visualizations per categories will be placed in this method as well. This method can call the two other ones which would actually align with what we do in other cases when giving "the user" some public function.